### ISSUE TITLE
Refactor machine_spec to 1) user_spec and 2) machine_spec. 

### ISSUE THREAD

**Issue by jmills-ncar** [NCAR/wrf_hydro_tests#30]
User specifications should be removed from the machine spec such that the machine spec can be used by all users on a given system

**Comment by jmccreight** [comment]
Related to #36 : where to stash the default PBS header? Probably in the user spec file.

User Spec:
default location: `~/.wrf_hydro_tests_user_spec.sh`
But can be specified in candidate spec.
Variables: 
```
export WRF_HYDRO_TESTS_DIR=/glade/u/home/jamesmcc/WRF_Hydro/wrf_hydro_tests
export GITHUB_SSH_KEY=''
export GITHUB_AUTHTOKEN=`cat ~/.github_authtoken 2> /dev/null`
export GITHUB_USERNAME=jmccreight

##PBS>>>>>>>>>>>>>>>>.
## Edit to your needs
#PBS -m abe 
#PBS -M `whoami`@ucar.edu
#PBS -A NRAL0017

## DO NOT TOUCH >>>> put these in a static place. Probably keep with qCleanCode.
#PBS -N $jobName
#PBS -l walltime=${wallTime}:00
#PBS -q $queue
#PBS -l select=${nNodesM1}:ncpus=36:mpiprocs=36+1:ncpus=${nCoresLeft}:mpiprocs=${nCoresLeft}
## Not using standard error and out files to capture model output
## but these hidden files might catch output and errors from the scheduler.
#PBS -o ${workingDir}/.${theDate}.pbs.stdout
#PBS -e ${workingDir}/.${theDate}.pbs.stderr
##<<<<<<<<<<<<<<<<<<<<PBS
```
Put flexibility to use_mpirun in user spec?
------------------------------

Machine Spec:
default location: `~/.wrf_hydro_tests_machine_spec.sh` ?? or in the repository (Later)
But can be specified in candidate spec.
Variables: 
```
if [[ $HOSTNAME == *cheyenne* ]]; then 
    if [[ $WRF_HYDRO_COMPILER == intel ]]; then
        export WRF_HYDRO_MODULES='intel/16.0.3 ncarenv/1.2 ncarcompilers/0.4.1 mpt/2.15f netcdf/4.4.1 nco/4.6.2 python/3.6.2'
    else 
        export WRF_HYDRO_MODULES='gnu/7.1.0 ncarenv/1.2 ncarcompilers/0.4.1 mpt/2.15 netcdf/4.4.1.1 nco/4.6.2 python/3.6.2'
    fi
fi
# Modules you want/need loaded on the machine.
# These are invoked all at once (order may matter) by `module load`. 
# Currently this includes modules needed for testing (e.g. on cheyenne: python and ncarenv (which contains nccmp))


export NETCDF=$(dirname `nc-config --includedir`)




function mpiRunFunc 
{ 
    local nCores=$1; 
    local theBinary=$2;
    dateTag=`date +'%Y-%m-%d_%H-%M-%S'`
    runCmd="mpirun -np $nCores ./`basename $theBinary` 1> ${dateTag}.stdout 2> ${dateTag}.stderr";
    echo $runCmd
    eval $runCmd 
    return $?
}
## This one covers docker. Works on cheyenne too, for small numbers of cores... for now.


function qCleanFunc 
{ 
    
    whtPath=`grep "wrf_hydro_tools=" ~/.wrf_hydro_tools | cut -d '=' -f2 | tr -d ' '`
    source $whtPath/utilities/sourceMe.sh || {
        echo "Your ~/.wrf_hydro_tools does not appear configured correctly."
        return 1
    }
    local nCores=$1; 
    local theBinary=$2;
    local jobName=$3
    local wallTime=$4
    runCmd="qCleanRun -j $jobName -W $wallTime $nCores ./`basename $theBinary`";
    echo $runCmd
    scriptOutput=`eval $runCmd`
    echo "$scriptOutput"
    jobId=`echo "$scriptOutput" | grep PBS_JOBID | cut -d' ' -f2 | cut -d'.' -f1`
    #echo jobId: $jobId
    jobStatus='Q'
    while [ "$jobStatus" != "F" ]; do
        sleep 5
        jobStatus=`qstat -x $jobId | tail -n1 | tr -s ' ' | cut -d' ' -f5`
        #echo "jobStatus: $jobStatus"
    done
    return 0
}

if [[ $HOSTNAME == *cheyenne* ]]; then 
    export -f qCleanFunc
    export WRF_HYDRO_RUN=qCleanFunc
else 
    export -f mpiRunFunc
    export WRF_HYDRO_RUN=mpiRunFunc
fi
```

**Comment by jmccreight** [comment]
Source machine spec first. Then source user spec. Let user spec override any variables in machine spec for flexibility.

**Comment by jmccreight** [comment]
closed with 4de3b501e4c492644268f9a80c85b6c96ee51d9f
